include/emp/Evolve/Systematics.hpp

```diff
@@ -1810,31 +1781,645 @@ namespace emp {
+    emp_optional_throw(anc_pos_it != header.end(),
+      "Input phylogeny file must be in ALife Phylogeny Data Standards format ancestor_list column is missing");
+    size_t anc_pos = std::distance(header.begin(), anc_pos_it);
+
+    auto origin_pos_it = std::find(header.begin(), header.end(), "origin_time");
+    int origin_pos = -1;
+    if(origin_pos_it != header.end()){
+      origin_pos = std::distance(header.begin(), origin_pos_it);
+    }
+
+    auto destruction_pos_it = std::find(header.begin(), header.end(), "destruction_time");
+    int destruction_pos = -1;
+    if (destruction_pos_it != header.end()) {
+      destruction_pos = std::distance(header.begin(), destruction_pos_it);
+    }
+
+    auto num_orgs_pos_it = std::find(header.begin(), header.end(), "num_orgs");
+    int num_orgs_pos = -1;
+    if (num_orgs_pos_it != header.end()) {
+      num_orgs_pos = std::distance(header.begin(), num_orgs_pos_it);
+    }
+
+    auto tot_orgs_pos_it = std::find(header.begin(), header.end(), "tot_orgs");
+    int tot_orgs_pos = -1;
+    if (tot_orgs_pos_it != header.end()) {
+      tot_orgs_pos = std::distance(header.begin(), tot_orgs_pos_it);
+    }
+
+    auto info_pos_it = std::find(header.begin(), header.end(), info_col);
+    emp_optional_throw(info_pos_it != header.end(),
+      "Input phylogeny file must be in ALife Phylogeny Data Standards format info column name supplied is not in file.");
+    size_t info_pos = std::distance(header.begin(), info_pos_it);
+
+    // Keep track taxon objects
+    std::unordered_map<int, emp::Ptr<taxon_t> > taxa;
+    // File is out of order, so we have to link up parents
+    // and offspring after the fact
+    std::unordered_map<int, int> unlinked_parents;
+    // Keep track of roots
+    emp::vector<emp::Ptr<taxon_t>> roots;
+
+    // Read in each row and make a taxon for it
+    size_t num_lines = in_file.GetNumLines();
+    for (size_t i = 0; i < num_lines; i++) {
+      emp::vector<std::string_view> row = in_file.ViewRowSlices(i);
+      int id = emp::from_string<int>(row[id_pos]);
+
+      // Inf means this taxon is still alive
+      // or we don't know which taxa are alive
+      std::string destruction_time = "inf";
+      if (destruction_pos != -1) {
+        destruction_time = row[destruction_pos];
+      } else {
+        destruction_time = "unknown";
+      }
+
+      ORG_INFO info;
+      if constexpr (std::is_same<ORG_INFO, std::string>::value) {
+        info = row[info_pos];
+      } else {
+        info = emp::from_string<ORG_INFO>(row[info_pos]);
+      }
+
+      // Load ancestor list
+      std::string ancestor_list_str = {row[anc_pos].begin(), row[anc_pos].end()};
+      emp::remove_chars(ancestor_list_str, "[]\"");
+      emp::Ptr<taxon_t> tax;
+
+      // Make taxon (parent is nullptr for now)
+      tax.New(id, info);
+      if (destruction_time != "inf") {
+        ancestor_taxa.insert(tax);
+      } else {
+        active_taxa.insert(tax);
```

Comment:
Oh, because they could be outside (i.e. non-extant, non-ancestor taxa? That's a good point. I don't think we necessarily want to prune them - it's okay to have outside taxa. But you're right that they should go in outside_taxa. Ugh, that's going to be a pain

Reply:
Hah, never mind, it wasn't that bad. Fixed!